vmdb/db/fixtures/ae_datastore/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/catalogbundleinitialization.rb

```diff
@@ -1,131 +1,115 @@
-      unless tag_value.blank?
-        $evm.log("info", "Adding tag_category:<#{tag_category.inspect}> value:<#{tag_value.inspect}> to Service:<#{service.name}>")
-        service.tag_assign("#{tag_category}/#{tag_value}")
-      end
-    end # if tags_regex
-  end # options_hash[0].each
-end
+  def override_service_name(dialogs_options_hash)
+    log_and_update_message(:info, "Processing override_service_name...", true)
 
-# Get the task object from root
-service_template_provision_task = $evm.root['service_template_provision_task']
-
-# Get destination service object
-service = service_template_provision_task.destination
-$evm.log("info", "Detected Service:<#{service.name}> Id:<#{service.id}>")
-
-# Get dialog options from options hash
-# I.e. {:dialog=>{"option_0_myvar"=>"myprefix", "option_1_vservice_workers"=>"2", "tag_0_environment"=>"test", "tag_0_location"=>"paris",
-# "option_2_vm_memory"=>"2048", "option_0_vlan"=>"Internal", "option_1_cores_per_socket"=>"1"}}
-dialog_options = service_template_provision_task.dialog_options
-$evm.log("info", "Inspecting Dialog Options:<#{dialog_options.inspect}>")
-
-# Get options_hash
-options_hash = get_options_hash(dialog_options)
-
-# Tag Parent Service
-tag_parent_service(service, options_hash)
-
-# Process Child Services
-service_template_provision_task.miq_request_tasks.each do |t|
-  # Child Service
-  child_service = t.destination
-  # Service Bundle Resource
-  child_service_resource = t.service_resource
-
-  # Increment the provision_index number since the child resource starts with a zero
-  group_idx = child_service_resource.provision_index + 1
-  $evm.log("info", "Child service name:<#{child_service.name}> group_idx:<#{group_idx}>")
-
-  # Create dialog options hash variable
-  dialog_options_hash = {}
-
-  # Set all dialog options pertaining to the catalog item plus any options destined for the catalog bundle
-  unless options_hash[0].nil?
-    unless options_hash[group_idx].nil?
-      # Merge child options with global options if any
-      dialog_options_hash = options_hash[0].merge(options_hash[group_idx])
-    else
-      dialog_options_hash = options_hash[0]
-    end
-  else # unless options_hash[0].nil?
-    unless options_hash[group_idx].nil?
-      dialog_options_hash = options_hash[group_idx]
+    new_service_name = dialogs_options_hash[0].fetch(:service_name, nil)  if dialogs_options_hash[0]
+    new_service_name = "#{@service.name}-#{Time.now.strftime('%Y%m%d-%H%M%S')}" if new_service_name.blank?
+    log_and_update_message(:info, "Setting Service name: #{new_service_name}")
+
+    @service.name = new_service_name
+
+    log_and_update_message(:info, "Processing override_service_name...Complete", true)
+  end
+
+  def override_service_description(dialogs_options_hash)
+    log_and_update_message(:info, "Processing override_service_description...", true)
+    new_service_description = dialogs_options_hash[0].fetch(:service_description, nil) if dialogs_options_hash[0]
```

Comment:
@tinaafitz 
The dialog_options_hash[0] is getting passed into this function so we don't have to call dialog_options_hash[0] again.
`override_service_name(dialog_options_hash.fetch(0, {}))`